Lesson20/ayjan/script.js

```diff
@@ -0,0 +1,496 @@
+    );
+    return newSavingsAcc;
+  }
+
+  createChecking({ ownerName, pin, initialBalance = 0, overdraftLimit = 200 }) {
+    const newCheckingAcc = new CheckingAccount(
+      ownerName,
+      pin,
+      initialBalance,
+      overdraftLimit
+    );
+    this.#accounts.set(newCheckingAcc.getAccountNumber(), newCheckingAcc);
+    console.log(
+      `Checking account created for ${ownerName} with account number ${newCheckingAcc.getAccountNumber()}.`
+    );
+    return newCheckingAcc;
+  }
+
+  findAccount(accountNumber) {
+    return this.#accounts.get(accountNumber) || null;
+  }
+
+  deposit(accountNumber, amount, description) {
+    const account = this.findAccount(accountNumber);
+    if (!account) {
+      console.error("Account not found.");
+      return;
+    }
+    account.deposit(amount, description);
+  }
+
+  withdraw(accountNumber, pin, amount, description) {
+    const account = this.findAccount(accountNumber);
+    if (!account) {
+      console.error("Account not found.");
+      return;
+    }
+    account.withdraw(amount, description, pin);
+  }
+
+  getBalance(accountNumber, pin) {
+    const account = this.findAccount(accountNumber);
+    if (!account) {
+      console.error("Account not found.");
+      return;
+    }
+    const balance = account.getBalance(pin);
+    if (balance !== undefined) {
+      console.log(`Balance for account ${accountNumber}: EUR${balance}`);
+    } else {
+      console.error("Unable to retrieve balance.");
+    }
+    return balance;
+  }
+
+  transfer(fromAccountNum, fromPin, toAccountNum, amount, description) {
+    const fromAccount = this.findAccount(fromAccountNum);
+    const toAccount = this.findAccount(toAccountNum);
+    if (!fromAccount || !toAccount) {
+      console.error("One or both accounts not found.");
+      return;
+    }
+    if(fromAccountNum === toAccountNum){
+      console.error("Cannot transfer to the same account.");
+      return;
+    }
+    const currentBalance = fromAccount.getBalance(fromPin);
+    if (currentBalance === undefined) {
+      console.error("Authentication failed for the source account.");
+      return;
+    }
+    if(currentBalance < amount && !(fromAccount instanceof CheckingAccount)){
+      console.error("Insufficient funds including overdraft limit.");
+      return;
+}
+console.log(`Initiating transfer...${amount} from ${fromAccountNum} to ${toAccountNum}`);
+  
+const transactionOut = new Transaction(
+  Transaction.TransactionType.transferOut,
+  amount,
+  description,
+  { to: toAccountNum }
+);
+fromAccount._commitTransaction(transactionOut);
+
+const transactionIn = new Transaction(
+  Transaction.TransactionType.transferIn,
+  amount,
+  description,
+  { from: fromAccountNum }
+);
+toAccount._commitTransaction(transactionIn);
+
+console.log("Transfer completed successfully.");
+
+
+}
+}
+
+const bank = new Bank();
+const acc1 = bank.createChecking({
```

Comment:
Try to always use a meaningful name that is easy to understand, e.g. checkingAccountAlice, savingsAccountBob, etc.